File: Campus smart hub/backend/models/Booking.js
const mongoose = require('mongoose');

const bookingSchema = new mongoose.Schema({
  resourceId: { type: mongoose.Schema.Types.ObjectId, ref: 'Resource', required: true },
  userId: { type: mongoose.Schema.Types.ObjectId, ref: 'User', required: true },
  date: { type: String, required: true }, // Format: YYYY-MM-DD
  startTime: { type: String, required: true }, // Format: HH:mm
  endTime: { type: String, required: true }, // Format: HH:mm
  purpose: { type: String, required: true },
  attendees: { type: Number, default: 1 },
  status: { 
    type: String, 
    enum: ['PENDING', 'APPROVED', 'REJECTED', 'CANCELLED'], 
    default: 'PENDING' 
  },
  adminReason: { type: String },
}, { timestamps: true });

module.exports = mongoose.model('Booking', bookingSchema);

File: backend/src/main/java/com/project/model/Booking.java
package com.project.model;

import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;
import org.springframework.data.annotation.CreatedDate;
import org.springframework.data.annotation.Id;
import org.springframework.data.mongodb.core.mapping.Document;

import java.time.LocalDateTime;
import java.util.List;

@Data
@Builder
@NoArgsConstructor
@AllArgsConstructor
@Document(collection = "bookings")
public class Booking {

    @Id
    private String id;
    
    private String resourceId;
    
    private String userId;
    
    private String date; // YYYY-MM-DD
    
    private String startTime; // HH:mm
    
    private String endTime; // HH:mm
    
    private String purpose;
    
    private Integer attendees;
    
    @Builder.Default
    private BookingStatus status = BookingStatus.PENDING;
    
    private String rejectionReason;

    private List<String> images;

    @CreatedDate
    private LocalDateTime createdAt;
}
